receiver/kubeletstatsreceiver/internal/kubelet/accumulator.go

```diff
@@ -71,10 +69,15 @@ func (a *metricDataAccumulator) nodeStats(s stats.NodeStats) {
 
 	startTime := pcommon.NewTimestampFromTime(s.StartTime.Time)
 	currentTime := pcommon.NewTimestampFromTime(a.time)
-	addCPUMetrics(ilm.Metrics(), nodePrefix, s.CPU, startTime, currentTime)
-	addMemoryMetrics(ilm.Metrics(), nodePrefix, s.Memory, currentTime)
-	addFilesystemMetrics(ilm.Metrics(), nodePrefix, s.Fs, currentTime)
-	addNetworkMetrics(ilm.Metrics(), nodePrefix, s.Network, startTime, currentTime)
+	addCPUMetrics(ilm.Metrics(), metadata.M.K8sNodeCPUUtilization, metadata.M.K8sNodeCPUTime, s.CPU, startTime,
+		currentTime)
+	addMemoryMetrics(ilm.Metrics(), metadata.M.K8sNodeMemoryAvailable, metadata.M.K8sNodeMemoryUsage,
+		metadata.M.K8sNodeMemoryRss, metadata.M.K8sNodeMemoryWorkingSet, metadata.M.K8sNodeMemoryPageFaults,
+		metadata.M.K8sNodeMemoryMajorPageFaults, s.Memory, currentTime)
```

Comment:
Do we really need to pass all the metadata definitions? It seems a bit clunky

Reply:
I agree, but it seemed like the way the reuse the most code.

We could get rid of the generic `add*Metrics` methods and call the more specific methods directly, passing in the exact init function, but then we'd have to do the `s == nil` check for each group of metrics for each of nodes, pods, and containers.  

We could also create object-specific methods, like a `addCPUMetricsForNode`, which would know to use the node metric initializers, but all the conditional checks would be duplicated between the different methods.

Reply:
We might be able to do something clever by making  `addCPUMetrics` a variadic function and passing in the init functions  via another function like 

```go
func getNodeCPUInit() []metadata.MetricIntf {
	return []metadata.MetricIntf {
		metadata.M.K8sPodCPUUtilization, 
		metadata.M.K8sPodCPUTime,
	}
}
```

This would make the call to `addCPUMetrics` cleaner and reduces the number of argument changes that would happen if we had a 3rd metric.

Downside is that now in `addCPUMetrics` you have to make assumptions about the order of the init functions, which I really dislike.

Reply:
You can add something like this to receiver/kubeletstatsreceiver/internal/metadata/metrics.go

```golang
type CpuMetrics struct {
	Time MetricIntf
	Utilization MetricIntf
}

var NodeCpuMetrics = CpuMetrics{
	Time: M.K8sNodeCPUTime
	Utilization: M.K8sNodeCPUUtilization
}

var PodCpuMetrics = CpuMetrics{
	Time: M.K8sPodCPUTime
	Utilization: M.K8sPodCPUUtilization
}
```

then pass `metadata.NodeCpuMetrics` or `metadata.PodCpuMetrics` instead

Reply:
I think that would work.  But if we add something to metadata/generated_metrics.go won't it be override?  Maybe accumulator.go or kubelet/metrics.go would be a better spot?  I would lean to accumulator since that is the first place it would get used.

Reply:
There will be enough code I could see it living it its own file as well. Something like definitions.go

Reply:
it has to be another file, name doesn't matter since it'll go away soon. I called it metrics.go :)

Reply:
The things we do for a refactor lol

Reply:
Done.  Much easier to read now.